### ISSUE TITLE
Appsync cannot sort items by string value with Elasticsearch

### ISSUE THREAD

**Issue by yezarela** [aws-amplify/amplify-cli#1683]
**Describe the bug**
I'm using amplify-cli to generate appsync resolvers with `@searchable` directive, but it seems that search resolver cannot sort items by `String` value, but it works with `AWSDateTime` value.

**To Reproduce**
Example request:
```
{
  searchUsers(sort: {
    field: fullName,
    direction: asc
  }) {
    items {
      id
      fullName
    }
  }
}
```

**Expected behavior**
User items sorted by fullName.

**Screenshots**
![image](https://user-images.githubusercontent.com/11919795/48695686-584fee80-ec12-11e8-8a8c-fbe81e64eb22.png)

**Desktop (please complete the following information):**
- Using AWS Console

**Smartphone (please complete the following information):**
 - Using AWS Console

**Additional context**
It also happened to my past appsync project.

***You can turn on the debug mode to provide more info for us by setting window.LOG_LEVEL = 'DEBUG'; in your app.***

**Comment by sacrampton** [comment]
@hakimio had the right approach. However, you have to search the data type in ElasticSearch. For example, I might have photographTaken : AWSDateTime in appsync which gets stored as String in DynamoDB and date in ElasticSearch. The one that matters is how its stored in ElasticSearch and I was struggling to be able to determine the data type in ElasticSearch in the resolver.

So the simplest and most reliable way was to specify if it was a string or non-string in the query. The way for me to achieve this was to modify the arguments of the search resolver to ask for a string field. If its a string enter "keyword", if its not a string then enter "none"

`query searchPhotos{
  searchPhotos(
    sort:{
      string: none
      direction: asc
      field: photographTaken
    }
    limit:10
  )}`

Then just have the resolver check if $context.args.sort.string == "keyword" and if it does add "${context.args.sort.field}.keyword" and if it does not just use "${context.args.sort.field}" as per @hakimio  instructions.

**Comment by SwaySway** [comment]
Closing this as this issue was addressed and resolved in #800.
There is  also a enhancement issue, #2517, regarding the change in types between appsync and elasticsearch.

Should there be in another questions related to this please feel free to comment.
